File: prisma/refs/py/minimum-interval-to-include-each-query.py
import heapq


def min_interval(intervals, queries):
    ivs = sorted(intervals)
    out = [-1] * len(queries)
    heap = []  # (size, right)
    idx = 0
    for q, qi in sorted((q, i) for i, q in enumerate(queries)):
        while idx < len(ivs) and ivs[idx][0] <= q:
            l, r = ivs[idx]
            heapq.heappush(heap, (r - l + 1, r))
            idx += 1
        while heap and heap[0][1] < q:
            heapq.heappop(heap)
        if heap:
            out[qi] = heap[0][0]
    return out

File: prisma/refs/java/minimum-interval-to-include-each-query.java
import java.util.*;

class Solution {
    public int[] minInterval(int[][] intervals, int[] queries) {
        int[][] ivs = intervals.clone();
        Arrays.sort(ivs, (a, b) -> a[0] - b[0]);
        Integer[] qi = new Integer[queries.length];
        for (int i = 0; i < queries.length; i++) qi[i] = i;
        Arrays.sort(qi, (a, b) -> queries[a] - queries[b]);
        int[] out = new int[queries.length];
        Arrays.fill(out, -1);
        PriorityQueue<int[]> heap = new PriorityQueue<>((a, b) -> a[0] - b[0]); // [size, right]
        int idx = 0;
        for (int i : qi) {
            int q = queries[i];
            while (idx < ivs.length && ivs[idx][0] <= q) {
                heap.add(new int[]{ivs[idx][1] - ivs[idx][0] + 1, ivs[idx][1]});
                idx++;
            }
            while (!heap.isEmpty() && heap.peek()[1] < q) heap.poll();
            if (!heap.isEmpty()) out[i] = heap.peek()[0];
        }
        return out;
    }
}
